sklearn/svm/src/newrand/newrand.h

```diff
@@ -5,43 +5,34 @@
      libsvm and liblinear.
      Sylvain Marie, Schneider Electric
      See <https://github.com/scikit-learn/scikit-learn/pull/13511#issuecomment-481729756>
-
  */
 #ifndef _NEWRAND_H
 #define _NEWRAND_H
 
 #ifdef __cplusplus
+#include <random>  // needed for cython to generate a .cpp file from newrand.h
 extern "C" {
 #endif
 
 // Scikit-Learn-specific random number generator replacing `rand()` originally
 // used in LibSVM / LibLinear, to ensure the same behaviour on windows-linux,
 // with increased speed
 // - (1) Init a `mt_rand` object
-#if INT_MAX == 0x7FFFFFFF
 std::mt19937 mt_rand(std::mt19937::default_seed);
-#elif INT_MAX == 0x7FFFFFFFFFFFFFFF
-std::mt19937_64 mt_rand(std::mt19937::default_seed);
-#else
-info("Random number generator is not fixed for this system. Please report issue. INT_MAX=%d\n", INT_MAX);
-exit(1);
-#endif
 
 // - (2) public `set_seed()` function that should be used instead of `srand()` to set a new seed.
 void set_seed(unsigned custom_seed) {
     mt_rand.seed(custom_seed);
 }
 
 // - (3) New internal `bounded_rand_int` function, used instead of rand() everywhere.
-inline int bounded_rand_int(int orig_range) {
-    // "LibSVM / LibLinear Original way" - make a 31bit or 63bit positive
+inline uint32_t bounded_rand_int(uint32_t range) {
+    // "LibSVM / LibLinear Original way" - make a 31bit positive
     // random number and use modulo to make it fit in the range
-    // return abs( (int)mt_rand()) % orig_range;
+    // return abs( (int)mt_rand()) % range;
 
     // "Better way": tweaked Lemire post-processor
     // from http://www.pcg-random.org/posts/bounded-rands.html
-    // TODO how could we make this casting safer, raising an error if lost information?
-    uint32_t range = uint32_t(orig_range);
     uint32_t x = mt_rand();
```

Comment:
Don't you think that it would be safer to keep this ? Indeed `bounded_rand_int` is marked as `inline` so I guess (but I'm no C++ expert) that the input can actually be anything, there is absolutely no protection. At least having this cast would ensure that when this code is badly used there are no side effects. What do you think ?

Reply:
Now that the input to bounded_rand_int is uint32_t, I think this extra step is redundant (casting uint32_t to uint32_t)

Reply:
ok !